src/components/ActionMenu/components/SecondaryAction/SecondaryAction.tsx

```diff
@@ -0,0 +1,33 @@
+import React, {useEffect, useRef} from 'react';
+
+import {Button} from '../../../Button';
+import type {ButtonProps} from '../../../Button';
+import {useFeatures} from '../../../../utilities/features';
+
+import styles from './SecondaryAction.scss';
+
+interface SecondaryAction extends ButtonProps {
+  getOffsetWidth?(width: number): void;
+}
+
+export function SecondaryAction({
+  children,
+  getOffsetWidth,
+  ...rest
+}: SecondaryAction) {
+  const {newDesignLanguage} = useFeatures();
+  const secondaryActionsRef = useRef<HTMLSpanElement>(null);
+
+  useEffect(() => {
+    if (!getOffsetWidth || !secondaryActionsRef.current || !newDesignLanguage)
+      return;
+
+    getOffsetWidth(secondaryActionsRef.current?.offsetWidth);
```

Comment:
This passes the rendered offset width back to the actions component. Much safer than adding refs within a loop in my opinion